src/core/ext/transport/binder/wire_format/binder_android.cc

```diff
@@ -243,36 +280,18 @@ absl::Status ReadableParcelAndroid::ReadBinder(
   return absl::OkStatus();
 }
 
-namespace {
-
-bool byte_array_allocator(void* arrayData, int32_t length, int8_t** outBuffer) {
-  std::string tmp;
-  tmp.resize(length);
-  *reinterpret_cast<std::string*>(arrayData) = tmp;
-  *outBuffer = reinterpret_cast<int8_t*>(
-      &(*reinterpret_cast<std::string*>(arrayData))[0]);
-  return true;
-}
-
-bool string_allocator(void* stringData, int32_t length, char** outBuffer) {
-  if (length > 0) {
-    // TODO(mingcl): Don't fix the length of the string
-    GPR_ASSERT(length < 100);  // call should preallocate 100 bytes
-    *outBuffer = reinterpret_cast<char*>(stringData);
-  }
-  return true;
-}
-
-}  // namespace
-
 absl::Status ReadableParcelAndroid::ReadByteArray(std::string* data) const {
-  return AParcel_readByteArray(parcel_, data, byte_array_allocator) == STATUS_OK
-             ? absl::OkStatus()
-             : absl::InternalError("AParcel_readByteArray failed");
+  std::vector<uint8_t> vec;
+  if (AParcelReadVector(parcel_, &vec) == STATUS_OK) {
+    data->resize(vec.size());
+    memcpy(&((*data)[0]), &vec[0], vec.size());
```

Comment:
Can we use `vec.data()` here?

Reply:
Done. Also added a guard to skip memcpy when size is 0